sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -132,8 +127,9 @@ object QueryExecutionErrors {
   }
 
   def simpleStringWithNodeIdUnsupportedError(nodeName: String): Throwable = {
-    new SparkUnsupportedOperationException(errorClass = "UNSUPPORTED_SIMPLE_STRING_WITH_NODE_ID",
-      messageParameters = Array(nodeName))
+    new SparkUnsupportedOperationException(
+      errorClass = "UNSUPPORTED_FEATURE",
+      messageParameters = Array(s"$nodeName does not implement simpleStringWithNodeId"))
```

Comment:
this does not sound like a user-facing error. Can we double-check it?

Reply:
Seems, it is not. At least, I haven't found a way how to trigger it from user level. I will replace the exception by `IllegalStateException` as for https://github.com/apache/spark/pull/35302#discussion_r792336423.